tobac/utils/general.py

```diff
@@ -631,6 +631,228 @@ def combine_feature_dataframes(
+    default: None | float = None,
+    id_column: str = "feature",
+) -> pd.DataFrame:
+    """
+    Get bulk statistics for objects (e.g. features or segmented features) given a labelled mask of the objects
+    and any input field with the same dimensions.
+
+    The statistics are added as a new column to the existing feature dataframe. Users can specify which statistics are computed by
+    providing a dictionary with the column name of the metric and the respective function.
+
+    Parameters
+    ----------
+    labels : np.ndarray[int]
+        Mask with labels of each regions to apply function to (e.g. output of segmentation for a specific timestep)
+    *fields : tuple[np.ndarray]
+        Fields to give as arguments to each function call. Must have the same shape as labels.
+    features: pd.DataFrame
+        Dataframe with features or segmented features (output from feature detection or segmentation)
+        can be for the specific timestep or for the whole dataset
+    func_dict: dict[str, Callable], optional (default: {'ncells':np.count_nonzero})
+        Dictionary with function(s) to apply over each region as values and the name of the respective statistics as keys
+        default is to just count the number of cells associated with each feature and write it to the feature dataframe
+    index: None | list[int], optional (default: None)
+        list of indexes of regions in labels to apply function to. If None, will
+            default to all integers between the minimum and the maximum value in labels
+    default: None | float, optional (default: None)
+        default value to return in a region that has no values
+    id_column: str, optional (default: "feature")
+       Name of the column in feature dataframe that contains IDs that match with the labels in mask. The default is the column "feature".
+
+     Returns:
+     -------
+     features: pd.DataFrame
+         Updated feature dataframe with bulk statistics for each feature saved in a new column
+    """
+    # raise error if mask and input data dimensions do not match
+    for field in fields:
+        if labels.shape != field.shape:
+            raise ValueError("Input labels and field do not have the same shape")
+
+    # mask must contain positive values to calculate statistics
+    if labels[labels > 0].size > 0:
+        if index is None:
+            index = range(
+                int(np.nanmin(labels[labels > 0])), int(np.nanmax(labels) + 1)
+            )
+        else:
+            # get the statistics only for specified feature objects
+            if np.max(index) > np.max(labels):
+                raise ValueError("Index contains values that are not in labels!")
+
+        # Find which labels exist in features for output:
+        index_in_features = np.isin(index, features[id_column])
+
+        # set negative markers to 0 as they are unsegmented
+        labels[labels < 0] = 0
+        bins = np.cumsum(np.bincount(labels.ravel()))
```

Comment:
To avoid changing the labels array:
```suggestion
        bins = np.cumsum(np.bincount(np.maximum(labels.ravel(), 0)))
```